awslimitchecker/services/cognito.py

```diff
@@ -0,0 +1,134 @@
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU Affero General Public License for more details.
+
+    You should have received a copy of the GNU Affero General Public License
+    along with awslimitchecker.  If not, see <http://www.gnu.org/licenses/>.
+
+The Copyright and Authors attributions contained herein may not be removed or
+otherwise altered, except to add the Author attribution of a contributor to
+this work. (Additional Terms pursuant to Section 7b of the AGPL v3)
+################################################################################
+While not legally required, I sincerely request that anyone who finds
+bugs please submit them at <https://github.com/jantman/awslimitchecker> or
+to me via email, and that you send any contributions or improvements
+either as a pull request on GitHub, or to me via email.
+################################################################################
+
+AUTHORS:
+Jason Antman <jason@jasonantman.com> <http://www.jasonantman.com>
+################################################################################
+"""
+
+import abc  # noqa
+import logging
+
+from .base import _AwsService
+from ..limit import AwsLimit
+from ..utils import paginate_dict
+
+logger = logging.getLogger(__name__)
+
+
+class _CognitoService(_AwsService):
+
+    service_name = "Cognito"
+    api_name = "cognito-idp"  # AWS API name to connect to (boto3.client)
+    quotas_service_code = "cognito-idp"
+
+    def find_usage(self):
+        """
+        Determine the current usage for each limit of this service,
+        and update corresponding Limit via
+        :py:meth:`~.AwsLimit._add_current_usage`.
+        """
+        logger.debug("Checking usage for service %s", self.service_name)
+        self.connect()
+        for lim in self.limits.values():
+            lim._reset_usage()
+        self.limits['User Pools']._add_current_usage(
+            len(
+                paginate_dict(
+                    self.conn.list_user_pools,
+                    MaxResults=60,
+                    alc_marker_path=['NextToken'],
+                    alc_data_path=['UserPools'],
+                    alc_marker_param='NextToken',
+                )["UserPools"]
+            ),
+            aws_type='AWS::Cognito::UserPool',
+        )
+        self._have_usage = True
+        logger.debug("Done checking usage.")
+
+    def _find_usage_user_pools(self):
+        """
+        List Cognito User Pools by calling AWS list_user_pools, and
+        update usage in self.limits for the following limits:
+
+        Global:
+        - User pools per account
+
+
+        TODO:
+        Per User Pool:
+        - App clients per user pool per user pool
+        - User import jobs per user pool
+        - Identity providers per user pool
+        - Resource servers per user pools
+        """
+
```

Comment:
This appears to be an empty method with no body at all. I'm guessing that you added this before putting the current user pools code directly in ``find_usage()``. If this method isn't going to be used, please remove it.